Services/Administration/OJS.Services.Administration.Business/Roles/RolesBusinessService.cs

```diff
@@ -0,0 +1,54 @@
+﻿namespace OJS.Services.Administration.Business.Roles;
+
+using Microsoft.EntityFrameworkCore;
+using OJS.Data.Models.Users;
+using OJS.Services.Administration.Data;
+using OJS.Services.Administration.Models.Roles;
+using OJS.Services.Infrastructure.Exceptions;
+using SoftUni.AutoMapper.Infrastructure.Extensions;
+using System;
+using System.Threading.Tasks;
+
+public class RolesBusinessService : AdministrationOperationService<Role, string, RoleAdministrationModel>, IRolesBusinessService
+{
+    private readonly IRoleDataService roleDataService;
+
+    public RolesBusinessService(IRoleDataService roleDataService)
+        => this.roleDataService = roleDataService;
+
+    public override async Task<RoleAdministrationModel> Get(string id)
+        => await this.roleDataService.GetByIdQuery(id).MapCollection<RoleAdministrationModel>().FirstAsync();
+
+    public override async Task<RoleAdministrationModel> Create(RoleAdministrationModel model)
+    {
+        var role = model.Map<Role>();
+
+        role.Id = Guid.NewGuid().ToString();
+        await this.roleDataService.Add(role);
+        await this.roleDataService.SaveChanges();
+
```

Comment:
Can you use RoleManager instead of dataService?

Reply:
We fixed this in users PR